### PR TITLE
OZ-457: Fixed issue with Nginx crashing when running the SENAITE service locally

### PR DESCRIPTION
**Pull request by Ruhanga**
Ticket: https://mekomsolutions.atlassian.net/browse/OZ-457

### CODE REVIEW

**Review by mks-d — COMMENTED** [review]
@Ruhanga in the headline of the PR, just say what it changes to the code base.
Don't speak of first, second, or follow-up commit, just summarise what it does.

File: scripts/start.sh [hunk 1]
```diff
@@ -33,7 +33,7 @@ INSTALLED_DOCKER_VERSION=$(docker version -f "{{.Server.Version}}")
 MINIMUM_REQUIRED_DOCKER_VERSION_REGEX="^((([2-9][1-9]|[3-9][0]|[0-9]{3,}).*)|(20\.([0-9]{3,}|[1-9][1-9]|[2-9][0]).*)|(20\.10\.([0-9]{3,}|[2-9][0-9]|[1][3-9])))"
 if [[ $INSTALLED_DOCKER_VERSION =~ $MINIMUM_REQUIRED_DOCKER_VERSION_REGEX ]]; then
     if command -v gp version &> /dev/null; then
-        export PROXY_TLS="-DgitPodEnvironment"
+        export SENAITE_GITPOD_HTTPS_REWRITE="true"
     fi
```
Comment (corneliouzbett) [inline scripts/start.sh:37]: Since we support both Traefik and Nginx, should we be checking if Traefik is enabled?

**Review by corneliouzbett — COMMENTED** [review]
I've tested this locally doesn't work for me. [Here](https://pastebin.mozilla.org/6Vvtpm7w) are ozone-proxy logs.

File: scripts/start.sh (hunk 1 above)
Comment (Ruhanga) [reply scripts/start.sh:37]: We've not yet run into issues with Traefik especially that we never run in with Gitpod. I'd worry about Traefik when its use within Gitpod shows up. We basically use traffic in prod.

Comment (Ruhanga) [reply scripts/start.sh]: I think the latest commit fixes the issue, https://github.com/ozone-his/ozone-docker-compose/pull/75/commits/1b11d6bd9df940285e99d19a111b23955e5e6836 .

Comment (Ruhanga) [reply scripts/start.sh]: Could you try again? Thanks.

**Review by corneliouzbett — COMMENTED** [review]
Tested locally works fine. A small comment, Otherwise LGTM!

File: scripts/start.sh [hunk 2]
```diff
@@ -33,7 +33,9 @@ INSTALLED_DOCKER_VERSION=$(docker version -f "{{.Server.Version}}")
 MINIMUM_REQUIRED_DOCKER_VERSION_REGEX="^((([2-9][1-9]|[3-9][0]|[0-9]{3,}).*)|(20\.([0-9]{3,}|[1-9][1-9]|[2-9][0]).*)|(20\.10\.([0-9]{3,}|[2-9][0-9]|[1][3-9])))"
 if [[ $INSTALLED_DOCKER_VERSION =~ $MINIMUM_REQUIRED_DOCKER_VERSION_REGEX ]]; then
     if command -v gp version &> /dev/null; then
-        export PROXY_TLS="-DgitPodEnvironment"
+        export SENAITE_GITPOD_HTTPS_REWRITE="true"
+    else
+        export SENAITE_GITPOD_HTTPS_REWRITE="false"
```
[the remark this answers is not in the document]
Comment (Ruhanga) [reply scripts/start.sh:38]: This was more like a bug fix that's specific to SENAITE. I'm not sure of other valid use-cases I have to worry about. I'd make this generic at the time this logic is needed (the architectural plan could also change at the time).

[the remark this answers is not in the document]
Comment (corneliouzbett) [reply scripts/start.sh:38]: I meant the name `SENAITE_GITPOD_HTTPS_REWRITE` to something like `GITPOD_ENV`. For instance;
```bash
if command -v gp version &> /dev/null; then
      export GITPOD_ENV="true"
  else
      export GITPOD_ENV="false"
 fi
```
Then for nginx `default.conf.template`
```nginx
map $GITPOD_ENV $senaite_rewrite {
    "true" /VirtualHostBase/https/$host/senaite/VirtualHostRoot/;
    default /VirtualHostBase/http/$host/senaite/VirtualHostRoot/;
}
```
Now, If I want to add a different rewrite or a configuration that is specified to the GitPod environment, I can simply re-use the same env var `GITPOD_ENV`.

[the remark this answers is not in the document]
Comment (Ruhanga) [reply scripts/start.sh:38]: Ahh, ok. I get the idea. I'll apply the suggestion.

**Review by corneliouzbett — APPROVED** [review]
Thanks, Nathan!

**Pull request merged**